core/kazoo_number_manager/src/providers/knm_providers.erl

```diff
@@ -328,15 +333,21 @@ exec(Number, Action=delete) ->
     lager:debug("requested number features: ~p", [RequestedModules]),
     exec(Number, Action, RequestedModules);
 exec(Number, Action) ->
+    {AllowedRequests, DeniedRequests} = split_requests(Number),
+    lager:debug("allowing number features ~p", [AllowedRequests]),
+    case DeniedRequests =:= [] of
+        true -> exec(Number, Action, AllowedRequests);
+        false ->
+            lager:debug("denied number features ~p", [DeniedRequests]),
+            knm_errors:unauthorized()
```

Comment:
@k-anderson note WRT numbers which somehow got denied features activated
* before this: the allowed features would get saved then the denied features would get deleted automatically
* with this PR: it fails here, before attempting any providers/feature operations (ie returns 403 instead of save-then-delete)